### PR TITLE
[TorchTidy] Add pattern to detect if bias is enabled in conv2d followed by batchnorm2d

### PR DESCRIPTION
**Pull request by davidchencsl**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82261
* #82248
* __->__ #81941
* #81921

Summary: For cuda models only, if we have nn.Conv2d followed by nn.Batchnorm2d, we don't actually need to use bias=True in Conv2d as the math will cancel out the effect of bias

Test Plan: Added test in test_profiler.py

Differential Revision: [D38082643](https://our.internmc.facebook.com/intern/diff/D38082643)

### CODE REVIEW

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Review by robieta — APPROVED** [review]
LGTM

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

**Comment by davidchencsl** [comment]
@davidchencsl has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38082643).

File: torch/profiler/_pattern_matcher.py [hunk 1]
```diff
@@ -427,6 +434,46 @@ def match(self, event: _ProfilerEvent):
...
+            return False
+        event = event.children[-1]
+        if event.name() != "aten::add_":
+            return False
+        # This means bias=True
+        event = self.go_up_until(
+            event, lambda e: e.name().startswith("nn.Module: Conv2d"))
+        if not event:
+            return False
+        event = self.next_of(event)
```
Comment (ngimel) [inline torch/profiler/_pattern_matcher.py:471]: how would this work if forward is something like
```
x1=self.conv1(x)
x2=self.conv2(x)
x1=self.bn1(x1)
x2=self.bn2(x2)
```
Wouldn't next_of be just the next conv in this case?

Comment (davidchencsl) [reply torch/profiler/_pattern_matcher.py:471]: Ah. Right now I am going to narrow the scope of detection to Situation in nn.Sequential, or simpler scenarios, as we do not have tensor identities information (Only size and dtype) as of right now. But great point! I will add a todo

Comment (davidchencsl) [reply torch/profiler/_pattern_matcher.py:471]: But if we do have that information, we can do something like:

event = next_used_event(event.output_tensor)

Comment (ngimel) [reply torch/profiler/_pattern_matcher.py:471]: tensor identities are needed for more robust pattern detection
